### DISCUSSION TITLE
Can I export a YOLOv8 model to only detects specific classes?

### DISCUSSION THREAD

**Discussion by gelbling** [ultralytics/ultralytics#4094] [Q&A] [answered]
Hello, I was wondering if I could export a YOLOv8 model that only detects specific classes. 

For example, when doing inference with the ultralytics library in Python I can specify the classes I want it to detect:
`    results = model(frame, classes=[2,3,7], save_conf=True, conf=0.25)  # return a list of Results objects
`

I'd like to know if this is possible with an exported model.

Thanks for the help!

**Comment by ExtReMLapin** [comment, marked as answer]
You can't "save space" as the neurons are linked to other classes, can you should be however to modifiy the head to only return the class you care about.

Or just filter results lol

**Reply by glenn-jocher** [reply]
@ExtReMLapin hello! When you export a YOLOv8 model, it retains the ability to detect all classes it was trained on. However, you can filter the detection results post-inference to only include the classes you're interested in, similar to how you're doing it in Python with the `classes` parameter. This filtering would need to be implemented in the post-processing step of your application after running the model. Keep in mind that the model's size and complexity remain unchanged, as it still contains all the information necessary to detect all trained classes. 😊👍
